float32  req
---
float32 rep
bool  repb
